File: 914/solution.go
package main

import "fmt"

func hasGroupsSizeX(deck []int) bool {
	X := len(deck)
	counts := make(map[int]int, X)
	for _, card := range deck {
		if _, exist := counts[card]; !exist {
			counts[card] = 1
		} else {
			counts[card]++
		}
	}

	var gcd func(int, int) int
	gcd = func(a, b int) int {
		if b == 0 {
			return a
		}
		return gcd(b, a%b)
	}

	for _, v := range counts {
		X = gcd(X, v)
	}
	return X >= 2
}

func main() {
	deck := []int{1, 2, 3, 4, 4, 3, 2, 1}
	fmt.Println(hasGroupsSizeX(deck))
}

File: 914/Solution.cpp
class Solution {
public:
    bool hasGroupSizeX(vector<int>& deck) {
        unordered_map<int, int> counts;
        for (int card : deck) ++counts[card];
        for (int i = 2; i <= deck.size(); ++i) {
            if (deck.size() % i) continue;
            bool ok = true;
            for (const auto& pair : counts) {
                if (pair.second % i) {
                    ok = false;
                    break;
                }
            }
            if (ok) return true;
        }
        return false;
    }
};

class Solution {
public:
    bool hasGroupSizeX(vector<int>& deck) {
        unordered_map<int, int> counts;
        for (int card : deck) ++counts[card];
        int X = deck.size();
        for (const auto& p : counts)
            X = __gcd(X, p.second);
        return X >= 2;
    }
}